mucommander-core/src/main/java/com/mucommander/ui/action/impl/InternalEditAction.java

```diff
@@ -70,7 +72,13 @@ public InternalEditAction(MainFrame mainFrame, Map<String,Object> properties) {
      */
     @Override
     protected void performInternalAction(AbstractFile file) {
-        EditorRegistrar.createEditorFrame(mainFrame, file, getIcon().getImage());
+        if (file.isDirectory()) {
+            FileSet fileSet = new FileSet();
+            fileSet.add(file);
+            new ChangePermissionsDialog(mainFrame, fileSet).showDialog();
+        } else {
+            EditorRegistrar.createEditorFrame(mainFrame, file, getIcon().getImage());
+        }

```

Comment:
don't we want this to happen also when custom-edit/view operation is defined? I'd say we should have the logic for folders in AbstractViewerAction#performAction. what do you think?

Reply:
hmm, but why? If someone overrides it with a custom action, why do we want to enforce some internal logic there?

Reply:
I think it depends on how the custom action is configured
if we use, let's say, Emacs as a viewer then there should be no problem to provide it with a folder
but that's a differently story in case we use something else that fails on folders

that said, it's not like someone lose functionality by defining a custom edit/view command - in that case, the edit/view actions on folders that we default to when no custom edit/view command is set can be invoked using their original shortcuts.

so I have no strong opinion on that, I'm ok with that if you think it's better that way